### PR TITLE
Update and test the `getting-started` script

### PR DESCRIPTION
**Pull request by rzadp**
Here are some changes to the `getting-started.sh` scripts we have advertised on top of the readme.

### Changes to the script

1. Change `echo` to a more portable `printf`.

On my machine, the script printed a literal `\n` string if run with `bash`.
If I changed it to `echo -e`, then it printed a literal `-e` if run with `sh`.

Changed it to `printf` which is more portable.

---

2. Template selection

The script proceeded to clone and build the `minimal` template, which is not always what we want.
Added a selection prompt where the user can select one of the 3 templates, and choose if it should be built&run or not.
The user can also select no template at all - that way, we have a starter of a dependencies-installation script.

---

3. Added some missing dependencies for some of the systems.

### A workflow testing the script

I propose a workflow, that will test the script using the [expect](https://core.tcl-lang.org/expect/index) tool.
For each OS mentioned in the script (macOS, Ubuntu, Debian, Arch, Fedora, OpenSUSE) we go through the script twice, and after that build and run the template binary.

I'm using docker containers, so we start from scratch and make sure the scripts installs all required dependencies - with the exception of macOS, which I can't run from scratch in a container.

The jobs use a selected combination of OSes, shell interpreters (`bash` or `sh`), and templates.
There is too much combinations to run them all, but I have [run it once](https://github.com/paritytech-stg/polkadot-sdk/actions/runs/10509533645) in staging to make sure all pass.

I'm adding a cron schedule because it can break without any code changes in this repository (e.g. new `latest` release of a container).

### CODE REVIEW

**Comment by kianenigma** [comment]
> Template selection

In this section, when they cancel, can you also make sure to show them "if you want more templates, please see [here](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/polkadot_sdk/templates/index.html)"? 

> A workflow testing the script


I am not an expert to judge the complexity of the testing scripts. I can share the overall guidance that we should also not over-engineer it, both to minimize the CI costs and the maintenance of the scripts. 

That is not to say the current steps are over-engineered per-se, this is more of an overall thought. 

What I would suggest is only to remove the last two steps: The templates are already checked to compile, run and produce blocks elsewhere. We don't need it here again. 

As noted elsewhere, if we can use the same getting started script to setup our CI environment, it will be a double win, and more incentive to battle-test it.

**Review by kianenigma — APPROVED** [review]
Some comments, but nothing major.

**Comment by rzadp** [comment]
> What I would suggest is only to remove the last two steps: The templates are already checked to compile, run and produce blocks elsewhere. We don't need it here again.

Hmm, I'm not sure we're on the same page here.
Yes, the templates are checked to compile, run and produce blocks elsewhere, but only on Ubuntu/Debian and macOS.

What this introduces, is making sure that those exact lists of dependencies and Rust setup is enough to build a template on a clean arch/fedora/opensuse.

But yeah, maybe this is not a place for it. 🤔

**Review by EgorPopelyaev — APPROVED** [review]

File: .github/workflows/check-getting-started.yml [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set additional expect flags if necessary
+        run: |
+          # Add a debug flag to expect, if github is re-run with debug logging enabled.
+          [ "${{ runner.debug }}" = "1" ] && EXPECT_FLAGS="-d" || EXPECT_FLAGS=""
+          echo "EXPECT_FLAGS=${EXPECT_FLAGS}" >> $GITHUB_ENV
+
+      - name: Check the first run if the script
```
Comment (ggwpez) [inline .github/workflows/check-getting-started.yml:90]:
```suggestion
      - name: Check the first run of the script
```

File: .github/workflows/check-getting-started.yml [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+          expect "compile the node?" {
+            send "n\r"
+          }
+
+          expect eof
+          '
+        timeout-minutes: 15
+
+      - name: Check the second run if the script
```
Comment (ggwpez) [inline .github/workflows/check-getting-started.yml:132]:
```suggestion
      - name: Check the second run of the script
```

File: .github/workflows/check-getting-started.yml [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
```
Comment (ggwpez) [inline .github/workflows/check-getting-started.yml:243]:
```suggestion
      - name: Check the second run of the script
```

**Review by ggwpez — APPROVED** [review]
Nice, thanks!

File: .github/workflows/check-getting-started.yml [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set additional expect flags if necessary
+        run: |
+          # Add a debug flag to expect, if github is re-run with debug logging enabled.
+          [ "${{ runner.debug }}" = "1" ] && EXPECT_FLAGS="-d" || EXPECT_FLAGS=""
+          echo "EXPECT_FLAGS=${EXPECT_FLAGS}" >> $GITHUB_ENV
+
+      - name: Check the first run if the script
```
Comment (rzadp) [inline .github/workflows/check-getting-started.yml:201]:
```suggestion
      - name: Check the first run of the script
```

**Review by mordamax — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Review by mutantcornholio — APPROVED** [review]

**Pull request merged**